src/main/java/com/google/api/codegen/gapic/TestGapicProvider.java

```diff
@@ -0,0 +1,110 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.gapic;
+
+import com.google.api.codegen.transformer.Transformer;
+import com.google.api.codegen.viewmodel.SurfaceSnippetSetRunner;
+import com.google.api.codegen.viewmodel.ViewModelDoc;
+import com.google.api.tools.framework.model.Interface;
+import com.google.api.tools.framework.model.Model;
+import com.google.api.tools.framework.model.stages.Merged;
+import com.google.api.tools.framework.snippet.Doc;
+
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+/** An implementation of GapicProvider for automated tests */
+public class TestGapicProvider implements GapicProvider<Interface> {
```

Comment:
Actually, for this class in particular, I see that it is identical to the SurfaceGapicProvider class in my own PR. I'm going to rename my class to ViewModelGapicProvider so that it can just be used for both cases. 

The point about the terms still stands for other contexts, though.